lf-edge/edge-home-orchestration-go
# [DataStorage] Build configuration files dynamically

- Configuration files(configuration.toml and datastorage-device.yaml) are generated automatically
  if the SERVICE environment variable contains DataStorage
- Name in configuration files would be set to datastorage-{deviceID}

Signed-off-by: Taewan Kim <t25.kim@samsung.com>

# Description

Fixes #302 
Related to #306 

## Type of change


# How Has This Been Tested?
1. Delete configuration files in /var/edge-orchestration/datastorage/
2. Run edge-orchestration with SERVICE environment variable.
```
$ docker run -it --rm --privileged --network="host" --name edge-orchestration -e SERVICE=DataStorage -v /var/edge-orchestration/:/var/edge-orchestration/:rw -v /var/run/docker.sock:/var/run/docker.sock:rw -v /proc/:/process/:ro edge-orchestration:coconut
```
3. Check the device profile from edgex-core-metadata from a web browser
  ex) http://localhost:48081/api/v1/deviceprofile

### RESULT 
![image](https://user-images.githubusercontent.com/5847128/121346290-e884e900-c960-11eb-99ff-664595f7d481.png)

**Test Configuration**:
* Firmware version: Ubuntu 20.04
* Hardware: x86-64
* Edge Orchestration Release: Coconut

# Checklist:
